src/wayland/selection/offer.rs

```diff
@@ -55,9 +55,27 @@ impl<U: Clone + Send + Sync + 'static> OfferReplySource<U> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DataControlOffer {
+    Wlr(ZwlrDataControlOfferV1),
+    Ext(ExtDataControlOfferV1),
+}
+
+impl DataControlOffer {
+    /// Advertise offered MIME type
+    ///
+    /// Sent immediately after creating the wlr_data_control_offer object. One event per offered MIME type.
+    fn offer(&self, mime_type: String) {
+        match self {
+            Self::Wlr(obj) => obj.offer(mime_type),
+            Self::Ext(obj) => obj.offer(mime_type),
+        }
+    }
+}
```

Comment:
I feel like it shouldn't be in this file, but rather in the place where you wrap data controls.

Reply:
Wrappers removed.